From 1d42842dbc90488d3ed40be79f80c7964df1da3e Mon Sep 17 00:00:00 2001
From: Eliad Peller <eliad@wizery.com>
Date: Mon, 1 Dec 2014 18:27:39 +0200
Subject: [PATCH 13/47] hostap: don't check scan results in case of 20mhz [AP
 AP]

When using ACS, we might get to ieee80211n_check_scan()
while having 20mhz configuration, which in turn might
cause unneeded channel swapping (e.g. 36 <-> 40).

Exit prematurely in this case.

Signed-off-by: Eliad Peller <eliad@wizery.com>
---
 src/ap/hw_features.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/ap/hw_features.c b/src/ap/hw_features.c
index 0b042b0..5d33940 100644
--- a/src/ap/hw_features.c
+++ b/src/ap/hw_features.c
@@ -292,6 +292,9 @@ static int ieee80211n_check_scan(struct hostapd_iface *iface,
 	int oper40;
 	int res;
 
+	if (!iface->conf->secondary_channel)
+		return 0; /* HT40 not used */
+
 	if (iface->current_mode->mode == HOSTAPD_MODE_IEEE80211A)
 		oper40 = ieee80211n_check_40mhz_5g(iface, scan_res);
 	else
-- 
1.9.1

